### PR TITLE
Add RedoCommand

### PR DESCRIPTION
**Pull request by gabrielchao**
Part of #59, #83.

- Add `redo` that can revert the previous data-modifying command if it was an `undo`.
- Add an `UndoCommandTest`.

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -265,11 +266,21 @@ public void discardStagedChanges() {
 
     @Override
     public List<HistoryRecord> revertTo(HistoryRecord record) throws NoSuchElementException {
```
Comment (cheongsiuhong) [inline src/main/java/seedu/address/model/ModelManager.java]: Javadoc would be good.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -265,11 +266,21 @@ public void discardStagedChanges() {
...
+        List<HistoryRecord> poppedRecords = historyManager.popRecordsTo(record, stagedAddressBook);
         changeBaseTo(record.getCopyOfAddressBook());
         return poppedRecords;
     }
 
+    @Override
+    public HistoryRecord redo() throws IllegalStateException {
+        Optional<HistoryRecord> redoneRecord = historyManager.popRedo(stagedAddressBook);
+        if (redoneRecord.isEmpty()) {
+            throw new IllegalStateException("Cannot redo: previous command was not an undo");
```
Comment (cheongsiuhong) [inline src/main/java/seedu/address/model/ModelManager.java]: Previous mutator command? Can I `redo` if I `undo` then use a non mutator command?

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (gabrielchao) [reply src/main/java/seedu/address/model/ModelManager.java]: Javadoc is in `Model.java`.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (gabrielchao) [reply src/main/java/seedu/address/model/ModelManager.java]: Yes, non `MutatorCommand`s will not affect undo/redo/history at all.

**Comment by gabrielchao** [comment]
> Generally looks good. Whats the interaction of `redo` and `revert` ?

It would only be possible to `redo` when the previous `MutatorCommand` was an `undo`.

**Comment by gabrielchao** [comment]
Merged as the changes since the review are minor.

**Pull request merged**